text/0750-feature-flag-advisor.md

```diff
@@ -0,0 +1,343 @@
+
+```
+> cdk flags 
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         - <unset>
+  
+  How would you like to proceed?
+   >  Set all flags to recommended values 
+      Set unconfigured flags to recommended values
+      Reset unconfigured flags to defaults (no impact)
+      Modify individual flag 
+```
+
+### Expected Use Cases
+
+#### View information about your feature flags
+
+To view a report of all flags, users can run `cdk flags --all`. This will display a report of every feature flag in the following
+order: feature flags with states that match our recommendations, feature flags with states that do not match our recommended values, and
+lastly, unconfigured feature flags.
+
+```
+> cdk flags --all
+    Feature Flag                              Recommended Value            User Value
+    @aws-cdk/...                              true                         true
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         - <unset>
+  
+  How would you like to proceed?
+   >  Set all flags to recommended values
+      Set all unconfigured flags to recommended values
+      Reset all unconfigured flags to defaults (no impact)
+      Modify individual flag
+```
+
+To view information about a specific flag, simply run `cdk flags "#FLAGNAME#"`.
+
+```
+> cdk flags "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021"
+    Feature Flag                                                        Recommended Value            User Value
+    "@aws-cdk/aws-cloudfront:defaultSecurityPolicyTLSv1.2_2021"         true                         true
+```
+
+#### Modify your feature flag values
+
+To modify your feature flag configurations, you can use the options `--set-recommended` and `--set-default`. In addition to these
+options, the `"#FLAGNAME#"` option also contains a `--set` suboption for the user to choose the new value.
+
+#### I want to change all my flags to the recommended states
+
+To change all your feature flags to our recommended states, you can run `cdk flags --set-recommended --all`.
+
+```
+> cdk flags --set-recommended --all
+    Feature Flag                              Recommended Value            User Value
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         false
+  * @aws-cdk/...                              true                         - <unset>
+  Setting all feature flags to the recommended values...
+
+  Resynthesizing... 
```

Comment:
At this point, did the context file already change? or are we still just proposing the change, not yet committing it to file? 

What if the user sees the proposed change and doesn't like it, how do they go back?

Reply:
Also I hope the plan here is to reuse `cdk diff`? because right now the output doesn't quite resemble it.